src/verdant/ghost-book/ghost-cell.tsx

```diff
@@ -42,71 +77,104 @@ class Cell extends React.Component<GhostCell_Props, { sample: string }> {
+        className={`${CONTAINER} ${active}`}
         onClick={() => this.props.clickEv()}
       >
-        <div className={`v-Verdant-GhostBook-cell-band ${active}`} />
-        <div className="v-Verdant-GhostBook-container-stack">
-          <div className="v-Verdant-GhostBook-cell-container">
-            <GhostCellLabel id={this.props.id} />
-            <div className={`v-Verdant-GhostBook-cell-content ${active}`}>
+        <div className={`${CELL_BAND} ${active}`}/>
+        <div className={CONTAINER_STACK}>
+          <div className={CELL_CONTAINER}>
+            <GhostCellLabel name={this.props.name} events={this.props.events}/>
+            <div className={`${CELL_CONTENT} ${active}`}>
               <div
-                className={`v-Verdant-GhostBook-cell ${this.props.name.charAt(
-                  0
-                )}  ${active}`}
-                dangerouslySetInnerHTML={{ __html: this.state.sample }}
-              ></div>
+                className={`${CELL} ${this.cellTypeCSS()}  ${active}`}
+                dangerouslySetInnerHTML={{__html: this.state.sample}}
+              />
             </div>
           </div>
-          {this.showOutput(nodey)}
+          {displayOutput ?
+            <GhostCellOutput
+              name={this.props.output}
+              codeCell={this.props.name}
+            /> : null}
         </div>
       </div>
     );
   }
 
+  private async updateSample() {
+    /* Update the sample HTML if it has changed */
+    let newSample = await this.getSample();
+    if (newSample.outerHTML != this.state.sample)
+      this.setState({sample: newSample.outerHTML});
+  }
+
   private async getSample() {
+    /* Get the new sample HTML */
     let nodey = this.props.history.store.get(this.props.name);
     if (!nodey) {
       // ERROR case
       console.log("ERROR: CAN'T FIND GHOST CELL", this.props.name);
+      return;
+    }
+    let diff;
+    if (this.props.diffPresent) {
+      diff = Sampler.PRESENT_DIFF;
+    } else if (this.props.events === undefined) {
+      diff = Sampler.NO_DIFF;
+    } else if (this.props.events.length === 0) { // optimizing case
+      diff = Sampler.NO_DIFF;
     } else {
-      let diff = this.props.events ? Sampler.CHANGE_DIFF : Sampler.NO_DIFF;
-      let sample = await VersionSampler.sample(
-        this.props.history,
-        nodey,
-        null,
-        diff
-      );
-      this.setState({ sample: sample.outerHTML });
+      diff = Sampler.CHANGE_DIFF;
     }
+    return VersionSampler.sample(
+      SAMPLE_TYPE.DIFF,
+      this.props.history,
+      nodey,
+      null,
+      diff,
+      this.props.prior
+    );
   }
 
-  private showOutput(nodey: Nodey) {
-    if (nodey instanceof NodeyCode && nodey.output)
-      return <GhostCellOutput id={this.props.output} />;
-    return null;
+  private cellTypeCSS(): string {
+    /* Returns the CSS class for the cell type */
+    switch (this.props.type) {
+      case CELL_TYPE.CODE:
```

Comment:
Why use an enum constant for type instead of checking against class name or nodey.typeChar?